lib/practice_exercises.rb

```diff
@@ -1,13 +1,58 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
+#
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+
+  array = string.split(//)
+
+  alphanumeric = /[\w]/i
+  least = 0
+  greatest = array.length - 1
+
+  while least < greatest
+
+    until alphanumeric.match(array[least])
+      least += 1
+    end
+    until alphanumeric.match(array[greatest])
+      greatest -= 1
+    end
+
+    if array[least].downcase != array[greatest].downcase
+      return false
+    end
+
+    least += 1
+    greatest -= 1
+  end
+
+  return true
+
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 def longest_prefix(strings)
```

Comment:
👍 However the time complexity is made up of two loops one through the longest string (m) and another through the list of strings (n).  So the time complexity is O(n * m).  

I would also call the space complexity O(m) as it's based on the size of the string.